src/sentry/migrations/0029_discover_query_upgrade.py

```diff
@@ -0,0 +1,150 @@
+        else:
+            field = convert_field(aggregate[0], False, False)
+        if field:
+            updated_query["fields"].append(field)
+
+    # Order by
+    orderby = saved_query.query.get('orderby', "")
+    unique = reverse = False
+    if orderby.startswith('-'):
+        reverse = True
+        orderby = orderby[1:]
+    if orderby.startswith('uniq_'):
+        unique = True
+        orderby = orderby[5:].replace('_', '.')
+    field = convert_field(orderby, unique, reverse)
+
+    if field:
+        updated_query['orderby'] = field
+        if reverse:
+            field = field[1:]
+        if field not in updated_query["fields"]:
+            updated_query["fields"].append(field)
+
+    # Conditions become a query now
+    for condition in saved_query.query.get("conditions", []):
+        column, operator, value = condition
+        if column in ['contexts.key']:
+            column = "tags[contexts.key]"
+        if column == "environment" and operator == "=":
+            updated_query['environment'].append(value.strip('"'))
+        elif operator == 'IS NOT NULL':
+            updated_query["query"].append(u"has:{}".format(column))
+        elif operator == 'IS NULL':
+            updated_query["query"].append(u"!has:{}".format(column))
+        elif column in SearchVisitor.numeric_keys or column in SearchVisitor.date_keys:
+            updated_query["query"].append(u"{}:{}{}".format(
+                column,
+                operator if operator != '=' else '',
+                value
+            ))
+        elif operator in ['LIKE', '=']:
+            updated_query["query"].append(u"{}:{}".format(column, prepare_value(value)))
+        elif operator in ['NOT LIKE', '!=']:
+            updated_query["query"].append(u"!{}:{}".format(column, prepare_value(value)))
+    updated_query["query"] = ' '.join(updated_query["query"])
+
+    # Create the version 2 query
+    new_query = DiscoverSavedQuery.objects.create(
+        organization=saved_query.organization,
+        name=saved_query.name + name_extra,
+        query=updated_query,
+        version=2,
+    )
+
+    # Set project_ids
+    existing_project_ids = DiscoverSavedQueryProject.objects.filter(
+        discover_saved_query=saved_query
+    ).values_list("project", flat=True)
+    new_query.set_projects(existing_project_ids)
+
+    return new_query
+
+def migrate_v1_queries(apps, schema_editor):
+    """
+    Creates v2 versions of existing v1 queries
+    """
+
+    """ Seq Scan on sentry_discoversavedquery
+        (cost=0.00..102.86 rows=1601 width=284)
+        (actual time=0.027..1.158 rows=1275 loops=1)
+        Filter: (version = 1)
+        Rows Removed by Filter: 69
+        Planning time: 0.929 ms
+        Execution time: 1.296 ms
+    """
+    queryset = DiscoverSavedQuery.objects.filter(version=1)
+
+    for query in RangeQuerySetWrapperWithProgressBar(queryset):
+        new_query = convert(query)
```

Comment:
`new_query` variable seems to be unused here. 👀 Does this do anything?

You don't need to fix this; I was just curious.

Reply:
Ah woops, that's a relic from me running tests on the migration... I'm... not going to fix this cause I don't want to wait for Travis to pass again 🙃